userspace/libsinsp/parsers.cpp

```diff
@@ -654,7 +661,51 @@ bool sinsp_parser::reset(sinsp_evt *evt) {
 		evt->get_tinfo()->m_flags |= PPM_CL_ACTIVE;
 	}
 
-	if(PPME_IS_ENTER(etype)) {
+	// Logic for new events
+	if(evt->is_syscall_event() && evt->is_new_event_version()) {
+		// todo!: check if we really need this logic, since in many other parts we check again the
+		// return value this could be a duplicate...or at least we can store it since we compute it
+		// for each event here.
+		if(evt->has_return_value()) {
+			int64_t res = evt->get_syscall_return_value();
+			if(res < 0) {
+				evt->set_errorcode(-(int32_t)res);
+			}
+		}
+
+		// The tinfo is never NULL here, we can move it outside at the end of the work
+		sinsp_threadinfo *tinfo = evt->get_tinfo();
+		ASSERT(tinfo);
+
+		if(evt->uses_fd()) {
+			// This is a strategy to save the fd even if we don't have the fdinfo available for that
+			// event. Probably we can do better and save the information in the event itself rather
+			// than in the thread. This a logic mainly used for enter and exit, not sure this would
+			// be still useful at the end of the work.
+			tinfo->m_lastevent_fd = evt->get_used_fd();
+
+			// todo!: note that `tinfo->get_fd` can return NULL if we dropped some events.
+			// if we look at many parsers we return immediately if `m_fdinfo` is NULL maybe we could
+			// return here (?)
+			evt->set_fd_info(tinfo->get_fd(tinfo->m_lastevent_fd));
+		}
+
+		if(evt->creates_fd()) {
+			// The fd is always the return value in this case.
+			int64_t res = evt->get_syscall_return_value();
```

Comment:
Perhaps we can cache this `res` since it is already called at L670, and avoid the doubled call.

Reply:
done, ty!